catalog/task/push-sbom-to-pyxis/0.1/push-sbom-to-pyxis.yaml

```diff
@@ -12,24 +12,90 @@ spec:
-      description: URL of the main binary image.
+    - name: imageURLs
+      description: Newline separated list of binary image URLs.
       type: string
+    - name: containerImageIDs
+      description: Newline separated list Pyxis image IDs
+      type: string
+    - name: pyxisSecret
+      type: string
+      description: The kubernetes secret to use to authenticate to Pyxis
+    - name: server
+      type: string
+      description: The server type to use. Options are 'production' and 'stage'
+      default: production
   volumes:
     - name: workdir
       emptyDir: {}
   steps:
-    - name: download-sbom-for-image
-      # Image taken from https://github.com/sigstore/cosign
-      # and copied to Quay
+    - name: download-sbom-files
+      image:
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      volumeMounts:
+        - mountPath: /workdir
+          name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        IMAGEURLS=($(params.imageURLs))
+        IMAGEIDS=($(params.containerImageIDs))
+
+        if [ ${#IMAGEURLS[@]} != ${#IMAGEIDS[@]} ]; then
+          echo "ERROR: Lengths of imageURLs and containerImageIDs do not match"
+          exit 1
+        fi
+
+        mkdir /workdir/sboms
+        cd /workdir/sboms
+
+        for i in ${!IMAGEURLS[@]}; do
+          echo "Fetching sbom for image: ${IMAGEURLS[$i]}"
+          cosign download sbom --output-file ${IMAGEIDS[$i]}.json ${IMAGEURLS[$i]}
+        done
+
+    - name: push-sbom-files-to-pyxis
       image:
-        quay.io/redhat-appstudio/cosign:v1.13.1
-      args:
-        - download
-        - sbom
-        - --output-file
-        - /workdir/sbom-cyclonedx.json
-        - "$(params.imageURL)"
+        quay.io/hacbs-release/release-utils:62f3c9ca85e45a4a76f2fb5c0ec9402f3cc08b2d
+      env:
+        - name: pyxisCert
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: cert
+        - name: pyxisKey
+          valueFrom:
+            secretKeyRef:
+              name: $(params.pyxisSecret)
+              key: key
       volumeMounts:
         - mountPath: /workdir
           name: workdir
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        if [[ "$(params.server)" == "production" ]]
+        then
+          PYXIS_URL="https://pyxis.api.redhat.com/"
+        elif [[ "$(params.server)" == "stage" ]]
+        then
+          PYXIS_URL="https://pyxis.preprod.api.redhat.com/"
+        else
+          echo "Invalid server parameter. Only 'production' and 'stage' are allowed."
+          exit 1
+        fi
+
+        echo "${pyxisCert}" > /workdir/crt
+        echo "${pyxisKey}" > /workdir/key
+        PYXIS_CERT_PATH=/workdir/crt
+        PYXIS_KEY_PATH=/workdir/key
+
+        cd /workdir/sboms
+
+        for FILE in *.json; do
+          # The upload_sbom script is not ready yet
+          # upload_sbom --sbom-file $FILE
```

Comment:
I guess we should probably wait for the script to be available before merging this PR? (That work is currently blocked by https://issues.redhat.com/browse/ISV-3314 )

Reply:
I am ok to merge this before the script is available if it helps us shake out any bugs along the way.

Reply:
keep in mind we do not have any customers yet onboarded...

Reply:
Yep, I'm ok with that.

Reply:
Never mind. I didn't manage to merge this before I left and now the `upload_sbom` is under review, so I will wait for that: https://github.com/redhat-appstudio/release-service-utils/pull/33